core/src/main/java/cz/xtf/core/openshift/ClusterVersionInfo.java

```diff
@@ -126,13 +129,149 @@ private String detectClusterVersionFromCluster() {
+            //   - The clients for OpenShift release 4.6.4:
+            // https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/4.6.4/
+            // - The latest client for a given OpenShift update channel; e.g.
+            //   - The latest in the 4.6 candidate channel:
+            // https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/candidate-4.6/
+            //   - The latest in the 4.5 stable channel:
+            // https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/stable-4.5/
+            // - The latest client for the channels of the latest GA OpenShift release.
+            //   - The latest client for the most recent GA release's candidate channel:
+            // https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/candidate/
+            //   - The latest client for the most recent GA release's stable channel:
+            // https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/stable/
+            //
+            // If you are looking for the latest stable release's client, please use:
+            // https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/stable/
+
+            // list of fallbacks - channel defaults to stable if not set
+            for (String version : Stream.of(
+                    openshiftVersion,
+                    getMajorMinorOpenshiftVersion(),
+                    getConfiguredChannel() + "-" + getMajorMinorOpenshiftVersion(),
+                    getConfiguredChannel())
+                    .map(Object::toString)
+                    .collect(Collectors.toList())) {
+
+                // list of fallbacks ( ocp, ocp-dev-preview )
+                List<Pair<String, String>> channels = Arrays.asList(Pair.of(version, "ocp"),
+                        Pair.of(version, "ocp-dev-preview"));
+
+                for (Pair<String, String> channel : channels) {
+                    String downloadUrl = getOcp4DownloadUrl(channel.getLeft(), channel.getRight());
+                    int code = Https.httpsGetCode(downloadUrl);
+                    if (code >= 200 && code < 300) {
+                        this.openshiftVersion = openshiftVersion;
+                        this.openshiftVersionURL = downloadUrl;
+                        return;
+                    }
+                }
+            }
+
+            // fallback - try stable
+
+            // list of fallbacks
+            List<Pair<String, String>> channels = Arrays.asList(Pair.of("stable", "ocp"), Pair.of("stable", "ocp-dev-preview"));
+
+            for (Pair<String, String> channel : channels) {
+                String downloadUrl = getOcp4DownloadUrl(channel.getLeft(), channel.getRight());
+                int code = Https.httpsGetCode(downloadUrl);
+                if (code >= 200 && code < 300) {
+                    this.openshiftVersion = null;
+                    this.openshiftVersionURL = downloadUrl;
+                    return;
```

Comment:
@mnovak1 This code should not be strictly necessary and is questionable if we assume this as expected behavior... if everything else fails try a stable version.
This is also partly because of consistency because if the version doesn't exist and the channel is not specified, it also defaults to stable [/core/src/main/java/cz/xtf/core/config/OpenShiftConfig.java#L110](https://github.com/xtf-cz/xtf/blob/67ac5d29a39e557318124b402a70b6e95ad4aacc/core/src/main/java/cz/xtf/core/config/OpenShiftConfig.java#L110)